### ISSUE TITLE
Programatically create colourSchemes

### ISSUE THREAD

**Issue by neefrehman** [kemiljk/better-logos-figma#29]
Just had the thought that we might be able to turn the coloured svg into a black or white one on the fly, and so would only need to upload one of each image to the CMS. This should be much more maintainable, and easier for people to contribute to.

1. when a black/white filter is active, we'd have to replace the `fill="whatever"` with `"black"` or `"white"` in the svg markup before sending it to figma. This should be possible with some regex.
2. we'd need to update the layout grid thumbnails as well — harder as that's a URL not markup.  Maybe we could do it with css filters?

**Comment by kemiljk** [comment]
So on this note, I was thinking something similar regarding the grey background. Currently, if a logo is white as it’s “Original” colour, or pale, the UI doesn’t render it with a background so it’s not visible fully. If we could programmatically check the `fill=` and look for colours within a light spectrum (say `.startsWith(“#F” || “#E” || “#D”`) for example, could that work do you think?

**Comment by neefrehman** [comment]
yeah should possible! that's a great idea. we could use the same regex util for both cases.

i had a quick look at some of the svgs and as suspected there are some edge cases. for example the amazon logo has a `fill="white"` attribute in all versions, but on a `clip-path` element. so we'd need to detect the fill only for `path`s. even then i'm sure there will be more issues! will try to test it out a bit this week.

**Comment by neefrehman** [comment]
Not had any time to look into this properly (v busy new project onboarding at work), but will try over the weekend :)

**Comment by kemiljk** [comment]
No stress! 😎
